File: app/src/main/java/dev/voleum/ordermolder/viewmodels/AbstractChooserItemViewModel.java
package dev.voleum.ordermolder.viewmodels;

import android.icu.text.DecimalFormat;

import androidx.databinding.BaseObservable;

import dev.voleum.ordermolder.helpers.DecimalHelper;

abstract public class AbstractChooserItemViewModel extends BaseObservable {

    protected DecimalFormat df;

    protected AbstractChooserItemViewModel() {
        df = DecimalHelper.newMoneyFieldFormat();
    }
}

File: app/src/main/java/dev/voleum/ordermolder/viewmodels/AbstractChooserItemViewModel.kt
package dev.voleum.ordermolder.viewmodels

import androidx.databinding.BaseObservable
import androidx.databinding.Bindable
import dev.voleum.ordermolder.helpers.DecimalHelper
import dev.voleum.ordermolder.models.Obj

abstract class AbstractChooserItemViewModel<T : Obj>(open val item: T) : BaseObservable() {

    protected val df = DecimalHelper.moneyFieldFormat()

    val name: String
        @Bindable get() = item.toString()
}